.teamcity/src/main/kotlin/configurations/CompileAll.kt

```diff
@@ -1,30 +1,59 @@
 package configurations
 
+import configurations.CompileAll.BuildCacheType.PRODUCTION_BUILD_CACHE
 import model.CIBuildModel
 import model.Stage
 
-class CompileAll(model: CIBuildModel, stage: Stage) : BaseGradleBuildType(stage = stage, init = {
-    id(buildTypeId(model))
-    name = "Compile All"
-    description = "Compiles all production/test source code and warms up the build cache"
+class CompileAll(model: CIBuildModel, stage: Stage, buildCacheType: BuildCacheType = PRODUCTION_BUILD_CACHE) :
```

Comment:
I wonder if it is simpler to create a completely new class instead of trying to parameterize this one.